modules/swagger-core/src/main/java/io/swagger/v3/core/jackson/ModelResolver.java

```diff
@@ -933,11 +935,29 @@ protected void _addEnumProps(Class<?> propClass, Schema property) {
         final boolean useIndex = _mapper.isEnabled(SerializationFeature.WRITE_ENUMS_USING_INDEX);
         final boolean useToString = _mapper.isEnabled(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);
 
+
+        Optional<Method> jsonValueMethod = Arrays.stream(propClass.getMethods())
```

Comment:
@frantuma @joschi 
It might be a better idea to utilize `findJsonValueAccessor` from `BeanDescriptor` since it also supports `JsonValue` annotations specified in interfaces.

The current implementation fails on 

```
public interface ApiEnum {
    @JsonValue
    String jsonValue();
}
```

```
public enum ResponseType implements ApiEnum {
    Http("http"),
    Websocket("websocket");

    private String description;

    ResponseType(String description) {
        this.description = description;
    }

    @Override
    public String jsonValue() {
        return description;
    }
}
```

while we're able to retrieve the actual `jsonValueAccessor` via

```
var typeFactory = TypeFactory.defaultInstance();
var type = typeFactory.constructFromCanonical(ResponseType.class.getCanonicalName());
var beanDesc = Json.mapper().getSerializationConfig().introspect(type);
var accessor = beanDesc.findJsonValueAccessor();
 ```